lib/ansible/modules/cloud/azure/azure_rm_aks.py

```diff
@@ -0,0 +1,537 @@
+        dns_prefix=aks.dns_prefix,
+        kubernetes_version=aks.kubernetes_version,
+        tags=aks.tags,
+        linux_profile=create_linux_profile_dict(aks.linux_profile),
+        service_principal_profile=create_service_principal_profile_dict(aks.service_principal_profile),
+        provisioning_state=aks.provisioning_state,
+        agent_pool_profiles=create_agent_pool_profiles_dict(aks.agent_pool_profiles),
+        type=aks.type
+    )
+
+
+def create_linux_profile_dict(linuxprofile):
+    '''
+    Helper method to deserialize a ContainerServiceLinuxProfile to a dict
+    :param: linuxprofile: ContainerServiceLinuxProfile with the Azure callback object
+    :return: dict with the state on Azure
+    '''
+    return dict(
+        ssh_key=linuxprofile.ssh.public_keys[0].key_data,
+        admin_username=linuxprofile.admin_username
+    )
+
+
+def create_service_principal_profile_dict(serviceprincipalprofile):
+    '''
+    Helper method to deserialize a ContainerServiceServicePrincipalProfile to a dict
+    Note: For security reason, the service principal secret is skipped on purpose.
+    :param: serviceprincipalprofile: ContainerServiceServicePrincipalProfile with the Azure callback object
+    :return: dict with the state on Azure
+    '''
+    return dict(
+        client_id=serviceprincipalprofile.client_id
+    )
+
+
+def create_agent_pool_profiles_dict(agentpoolprofiles):
+    '''
+    Helper method to deserialize a ContainerServiceAgentPoolProfile to a dict
+    :param: agentpoolprofiles: ContainerServiceAgentPoolProfile with the Azure callback object
+    :return: dict with the state on Azure
+    '''
+    return [dict(
+        count=profile.count,
+        vm_size=profile.vm_size,
+        name=profile.name,
+        os_disk_size_gb=profile.os_disk_size_gb,
+    ) for profile in agentpoolprofiles]
+
+
+class AzureRMManagedCluster(AzureRMModuleBase):
+    """Configuration class for an Azure RM container service (AKS) resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            dns_prefix=dict(
+                type='str',
+                required=True
+            ),
+            kubernetes_version=dict(
+                type='str',
+                required=False
+            ),
+            linux_profile=dict(
+                type='list',
+                required=True
+            ),
+            agent_pool_profiles=dict(
+                type='list',
+                required=True
```

Comment:
For the item which has type as `list` and element type is `dict`, you can write the spec like
```
agent_pool_profiles=dict(
  type='list',
  elements='dict',
  options=sub_spec
)
```
to define the suboptions. E.G. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py#L374